src/main/java/reposense/parser/DateArgumentType.java

```diff
@@ -15,12 +15,12 @@
  */
 public class DateArgumentType implements ArgumentType<Optional<Date>> {
     private static final String PARSE_EXCEPTION_MESSAGE_INVALID_DATE_STRING_FORMAT = "Invalid Date: %s";
-    private static final DateFormat CLI_ARGS_DATE_FORMAT = new SimpleDateFormat("dd/MM/yyyy");
+    private static final DateFormat CLI_ARGS_DATE_FORMAT = new SimpleDateFormat("dd/MM/yyyy HH:mm:ss");
```

Comment:
May need to enforce user only enters `dd/MM/yyyy` and not `dd/MM/yyyy+`. 

Try the following:
```
java -jar RepoSense.jar --since "10/10/2018 10:10:10"
java -jar RepoSense.jar --since "10/10/2018 10:"
```

It seems to get parsed properly 🤔.

Reply:
Hmm, this problem exists even before this PR. Should I fix it in this PR or open an issue for this?

Reply:
I have read online stating that `setLenient(false)` does not achieve strict date formatting. Hence examples such as `--since 1/1/201` will still be parsed successfully.

Reply:
Would implementing a regex to check if the argument is exactly `dd/MM/yyyy` format be an ideal solution?

Reply:
> I have read online stating that setLenient(false) does not achieve strict date formatting.

I think the dates like `1/1/201` are okay.

However, my concern is the behavior of `--since "10/10/18 11:11:11 22:22:22 ... 00:00:00"` and `--since "10/10/10 11/11/11 12/12/12"` as they are considered valid 😂.

I think we should do enforce `dd/MM/yyyy` and fast fail.
@eugenepeh your opinion?

Reply:
>I think the dates like 1/1/201 are okay.

Its okay with me, so no need to open an issue for it.

Why the need to include time in cli argument though?

Reply:
The time is not included in the cli argument, but the cli argument `sinceDate` will be appended with time `00:00:00` while `untilDate` will be appended with `23:59:59` so that they are parsed into Date objects such that the `untilDate` is included in the analysis.

Reply:
I see, sounds good to me. However, just to be wary of users' accidental input, can you add a few tests with time to ensure that it is stable?

Reply:
You mean tests for accidental input with time such as `--since 11/11/2019 00:00:00`?

Reply:
@jamessspanggg 
Ya something to test our assumption. Take a look at `ArgsParserTest`.

`--since "10/10/18 11:11:11 22:22:22"` + `00:00:00` = `10/10/18 00:00:00`
`--since "10/10/18 11:11:11"` + `00:00:00` = `10/10/18 00:00:00`
`--since "10/10/18 11:"` + `00:00:00` = `10/10/18 00:00:00`
`--since "10/10/18 11"` + `00:00:00` = `10/10/18 00:00:00`

Reply:
>You mean tests for accidental input with time such as --since 11/11/2019 00:00:00?

Yes, maybe something like what @yong24s has illustrated in the first comment. 
E.g. `--since "10/10/2018 10:"`

Reply:
> E.g. `--since "10/10/2018 10:"`

Currently arguments
- `--since "10/10/2018 10:` = `10/10/2018 10:00:00`
- `--since "10/10/2018 11:11:11 12:12:12` = `10/10/2018 11:11:11`

Should these arguments be considered as invalid? or should we only get the date from the argument and append the time accordingly (eg. `00:00:00` for sinceDate, `23:59:59` for untilDate)

Reply:
Using --since with invalid dates without the double quotes should fail (and intended).

Using --since with the double quotes is not an intended option (double quotes isn't highlighted in the UG). So was wondering if this should still be a concern

Reply:
I have updated the parsing of `sinceDate` and `untilDate` such that only the first substring that matches the date regex of `dd/mm/yyyy` will be used.

For e.g. 
- `--since "10/10/2018 10:"` will be parsed with `sinceDate` of 10/10/2018, and will be appended with time of `00:00:00`
- `--until "10/10/2018 11/11/11 12/12/12"` will be parsed with `untilDate` of 10/10/2018, and will be appended with time of `23:59:59`

Reply:
👍 do add a unit test too.

Reply:
Added unit tests.

Reply:
@jylee-git 
> So was wondering if this should still be a concern

Valid point.

We are also considering the possibility of typos that may increase the time of analysis. i.e. `1/1/2019` -> `1/1/201`.

Usually, this will be a low priority but since its a easy fix for added robustness 🤭.